kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -146,71 +150,114 @@ public synchronized List<String> getInterfaces() throws DBusException {
+            Properties deviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME, device.get().getObjectPath(),
+                    Properties.class);
 
-            List<Properties> accessPoints = getAllAccessPoints(wirelessDevice);
+            DBusPath ip4configPath = deviceProperties.Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_IP4CONFIG);
+            Optional<Properties> ip4configProperties = Optional.empty();
 
-            DBusPath activeAccessPointPath = wirelessDeviceProperties.Get(NM_DEVICE_WIRELESS_BUS_NAME,
-                    "ActiveAccessPoint");
-            Optional<Properties> activeAccessPoint = Optional.empty();
+            if (!ip4configPath.getPath().equals("/")) {
+                ip4configProperties = Optional
+                        .of(this.dbusConnection.getRemoteObject(NM_BUS_NAME, ip4configPath.getPath(),
+                                Properties.class));
+            }
 
-            String countryCode = IwCapabilityTool.getWifiCountryCode(commandExecutorService);
-            List<WifiChannel> supportedChannels = IwCapabilityTool.probeChannels(interfaceName, commandExecutorService);
+            if (!STATUS_SUPPORTED_DEVICE_TYPES.contains(deviceType)) {
+                logger.warn("Device \"{}\" of type \"{}\" currently not supported", interfaceName, deviceType);
+                return null;
+            }
 
-            if (!activeAccessPointPath.getPath().equals("/")) {
-                activeAccessPoint = Optional.of(this.dbusConnection.getRemoteObject(NM_BUS_NAME,
-                        activeAccessPointPath.getPath(), Properties.class));
+            switch (deviceType) {
+                case NM_DEVICE_TYPE_ETHERNET:
+                    Wired wiredDevice = this.dbusConnection.getRemoteObject(NM_BUS_NAME, device.get().getObjectPath(),
+                            Wired.class);
+                    Properties wiredDeviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME,
+                            wiredDevice.getObjectPath(), Properties.class);
+
+                    DevicePropertiesWrapper ethernetPropertiesWrapper = new DevicePropertiesWrapper(deviceProperties,
+                            Optional.of(wiredDeviceProperties), NMDeviceType.NM_DEVICE_TYPE_ETHERNET);
+
+                    networkInterfaceStatus = NMStatusConverter.buildEthernetStatus(interfaceName,
+                            ethernetPropertiesWrapper, ip4configProperties);
+                    break;
+                case NM_DEVICE_TYPE_LOOPBACK:
+                    DevicePropertiesWrapper loopbackPropertiesWrapper = new DevicePropertiesWrapper(deviceProperties,
+                            Optional.empty(), NMDeviceType.NM_DEVICE_TYPE_LOOPBACK);
+
+                    networkInterfaceStatus = NMStatusConverter.buildLoopbackStatus(interfaceName,
+                            loopbackPropertiesWrapper, ip4configProperties);
+                    break;
```

Comment:
My OCD requires me to comment on the fact that, for consistency, I would add the respective `create` methods here:

```suggestion
                    networkInterfaceStatus = createWiredStatus(...);
                    break;
                case NM_DEVICE_TYPE_LOOPBACK:
                    networkInterfaceStatus = createLoopbackStatus(...);
                    break;
```

Reply:
IMHO, this class needs a refactor since it has too many responibilities. This suggestion can be done in that activity.